clinica/iotools/converters/oasis_to_bids/oasis_to_bids_utils.py

```diff
@@ -39,63 +35,37 @@ def create_sessions_dict(
         Session dict.
     """
 
-    location = f"{StudyName.OASIS.value} location"
-    sessions = pd.read_csv(clinical_specifications_folder / "sessions.tsv", sep="\t")
-    sessions_fields = sessions[StudyName.OASIS.value]
-    field_location = sessions[location]
-    sessions_fields_bids = sessions["BIDS CLINICA"]
-    fields_dataset = []
-    fields_bids = []
+    study = StudyName.OASIS.value
+    location = f"{study} location"
+    spec = pd.read_csv(clinical_specifications_folder / "sessions.tsv", sep="\t")[
+        [study, location, "BIDS CLINICA"]
+    ].dropna()
     sessions_dict = {}
 
-    for i in range(0, len(sessions_fields)):
-        if not pd.isnull(sessions_fields[i]):
-            fields_bids.append(sessions_fields_bids[i])
-            fields_dataset.append(sessions_fields[i])
-
-    for i in range(0, len(sessions_fields)):
-        # If the i-th field is available
-        if not pd.isnull(sessions_fields[i]):
-            # Load the file
-            tmp = field_location[i].split("/")
-            location = tmp[0]
-            sheet = tmp[1] if len(tmp) > 1 else 0
-            file_to_read_path = clinical_data_dir / location
-            file_ext = os.path.splitext(location)[1]
-            if file_ext == ".xlsx":
-                file_to_read = pd.read_excel(file_to_read_path, sheet_name=sheet)
-            elif file_ext == ".csv":
-                file_to_read = pd.read_csv(file_to_read_path)
-            else:
-                raise ValueError(
-                    f"Unknown file extension {file_ext}. Expecting either .xlsx or .csv."
-                )
-
-            for r in range(0, len(file_to_read.values)):
-                # Extracts the subject ids columns from the dataframe
-                subj_id = file_to_read.iloc[r]["ID"]
-                if hasattr(subj_id, "dtype"):
-                    if subj_id.dtype == np.int64:
-                        subj_id = str(subj_id)
-                # Removes all the - from
-                subj_id_alpha = str(subj_id[0:3] + "IS" + subj_id[3] + subj_id[5:9])
-
-                # Extract the corresponding BIDS id and create the output file if doesn't exist
-                subj_bids = [s for s in bids_ids if subj_id_alpha in s]
-                if subj_bids:
-                    subj_bids = subj_bids[0]
-                    subj_dir = bids_dir / subj_bids
-                    session_names = get_bids_sess_list(subj_dir)
-                    for s in session_names:
```

Comment:
I'm not sure I understand how you got rid of this loop on the sessions.

Reply:
I am not fully satisfied with what I did : since it's OASIS1 there is only one session, and it is hardcoded everywhere with "ses-M000" so I did the same

Reply:
Hmm, ok I see. It's good to know to understand what you did.